File: ProjetoWeb/src/app/shared/model/key.ts
import { Game } from "./game";
import User from "./user";

export class Key {

    id? :string;
    codigo: string;
    preco: Number;
    user_id: string;
    user_email: string;
    game_id: string;
    
    // user: User;
    // game: Game;
    constructor(codigo: string, preco: Number, user_id: string, user_email: string, game_id: string, id?: string) {
        this.id = id;
        this.codigo = codigo;
        this.preco = preco;
        this.user_id = user_id;
        this.user_email = user_email
        this.game_id = game_id;
        
        // this.user = user;
        // this.game = game;
    }
}

File: Pweb1-Backend/src/main/java/com/example/pweb1backend/model/Key.java
package com.example.pweb1backend.model;


import jakarta.persistence.*;

import java.math.BigDecimal;
import java.math.BigInteger;

@Entity
@Table(name = "tb_key")
public class Key {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String codigo;

    private BigDecimal preco;


    private String user_id;

    private String user_email;

    private String game_id;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCodigo() {
        return codigo;
    }

    public void setCodigo(String codigo) {
        this.codigo = codigo;
    }

    public BigDecimal getPreco() {
        return preco;
    }

    public void setPreco(BigDecimal preco) {
        this.preco = preco;
    }

    public String getUser_id() {
        return user_id;
    }

    public void setUser_id(String user_id) {
        this.user_id = user_id;
    }

    public String getUser_email() {
        return user_email;
    }

    public void setUser_email(String user_email) {
        this.user_email = user_email;
    }

    public String getGame_id() {
        return game_id;
    }

    public void setGame_id(String game_id) {
        this.game_id = game_id;
    }
}
